### ISSUE TITLE
amp-story: Missing or inaccurate page titles

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#32510]
# Missing or inaccurate page titles

## WCAG Level
Level A

## Priority
Low

## Pages/screens/components affected
* [Navigation](https://www.gstatic.com/amphtml/stamp/qa/navigation.html)
* [RTL](https://www.gstatic.com/amphtml/stamp/qa/rtl.html)
* [Video](https://www.gstatic.com/amphtml/stamp/qa/video.html)
* [Audio - amp-story background audio](https://www.gstatic.com/amphtml/stamp/qa/audio-story.html)
* [Audio - amp-story-page background audio](https://www.gstatic.com/amphtml/stamp/qa/audio-page.html)
* [Audio - amp-story using amp-audio](https://www.gstatic.com/amphtml/stamp/qa/amp-audio.html)
* [Animation](https://www.gstatic.com/amphtml/stamp/qa/animation.html)
* [Sharing](https://www.gstatic.com/amphtml/stamp/qa/sharing.html)
* [Tooltip](https://www.gstatic.com/amphtml/stamp/qa/tooltip.html)
* [Page attachment](https://www.gstatic.com/amphtml/stamp/qa/page-attachment.html)
* [Branching](https://www.gstatic.com/amphtml/stamp/qa/branching.html#page=page2)
* [Consent](https://www.gstatic.com/amphtml/stamp/qa/consent.html)
* [Prerender - Image](https://www.gstatic.com/amphtml/stamp/qa/prerender-img.html?amp_js_v=0.1#origin=https://www.google.com&prerenderSize=1&visibilityState=prerender&paddingTop=0&p2r=0&horizontalScrolling=0&csi=1&aoh=15526927198732&viewerUrl=https://www.google.com&history=1&storage=1&cid=1&cap=swipe,navigateTo,cid,fragment,replaceUrl)
* [Prerender - Video](https://www.gstatic.com/amphtml/stamp/qa/prerender-video.html?amp_js_v=0.1#origin=https://www.google.com&prerenderSize=1&visibilityState=prerender&paddingTop=0&p2r=0&horizontalScrolling=0&csi=1&aoh=15526927198732&viewerUrl=https://www.google.com&history=1&storage=1&cid=1&cap=swipe,navigateTo,cid,fragment,replaceUrl)
* [AMP Story Player - Player](https://www.gstatic.com/amphtml/stamp/qa/player.html)
* [AMP Story Player - AMP Story 360](https://www.gstatic.com/amphtml/stamp/qa/amp-story-360.html)

## Description
A page's title is generally displayed by browsers in the title bar, as a label for tabs, and as default name when saving a page as bookmark/favourite. Page titles are generally announced to users of assistive technologies when they switch to the browser, or switch between tabs. Page titles should be descriptive of the topic/content of the page, to aid users orient themselves.

All the tested pages currently have the same generic "QA Story" title, which does not convey the actual content of the specific stories.

The one exception is [AMP Story Player - Player](https://www.gstatic.com/amphtml/stamp/qa/player.html) which, due to the way it is constructed, lacks a page title altogether. This leads to failing not only [WCAG 2.1 Success Criterion 2.4.2 Page Titled (Level A)](https://www.w3.org/TR/WCAG21/#page-titled), but also [WCAG 2.1 Success Criterion 4.1.1 Parsing (Level A)](https://www.w3.org/TR/WCAG21/#parsing).

## User impact
If a page title is not sufficiently descriptive, users - particularly users of assistive technologies - may have difficulty orienting themselves within a site. If users are looking for a particular page, and the title of page that is announced when navigating to it is not descriptive enough, they may be unsure if they reached their desired destination until they spend more time exploring/reading the page. Further, when switching between different applications, or different tabs within the same browser, it may not be apparent to these users when they switched to the intended browser instance or tab.

Even though the tested pages are only (presumably internal) QA testing pages, they will still benefit from having slightly more descriptive titles - particularly when a tester has a few tests open in different tabs, they'll be able to more readily find a specific test they've been working on/evaluating.

## Required solution
The page title must reflect the primary content and purpose of the page. Ensure that each page has an individual page title which adequately summarizes the page content. For "single-page application" type pages, or where page content is dynamically changed as the user moves through various steps (e.g. a Wizard), make sure that the title is dynamically updated through scripting.

While not necessarily required, for consistency we'd recommend adopting a general pattern for page titles throughout a site. The specific pattern can vary depending on the complexity and particular needs of a site, but as a starting point, we suggest the following basic format:

```
<html ...>
  <head>
    ...
    <title>page name – section heading – site name</title>
    ...
  </head>
  ...
</html>
```

As a best practice, the page name should match - or should at least include - the page's main heading - generally, its `<h1>` element.

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Implementation guidance

We recommend that all QA tests use the short descriptive titles that were provided for this accessibility assessment, e.g. "Audio - amp-story background audio - QA Story" or similar.

In the case of the [AMP Story Player - Player](https://www.gstatic.com/amphtml/stamp/qa/player.html), the page itself currently lacks an actual `<title>` element.

```
<!doctype html>
<html>
<head>
  <script async src="..." nonce="..."></script>
  <link href="..." rel="stylesheet" type="text/css" nonce="..." />
  <meta name="viewport" content="...">
</head>
<body>
  <amp-story-player style="...">
    <a href="./player-story-1.html">
      Story Player QA document: Story 1
    </a>
    <a href="./player-story-2.html">
      Story Player QA document: Story 2
    </a>
    <a href="./player-story-3.html">
      Story Player QA document: Story 3
    </a>
  </amp-story-player>
</body>
</html>
```

The individual sub-pages (which are loaded into the player dynamically) do all have a `<title>` (though it is, once more, the generic "QA Story" title). In this case, we would recommend dynamically injecting the `<title>` on the player page, to reflect the currently displayed story.

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Check that each page - whether a separate HTML document or different "view" that appears as a distinct page in a single-page application/site - has a title in the `<title>` element of the document's `<head>` section.
2. Check that the title is relevant and appropriate for the content of the page/view.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website has been tested for the same issue.
* All issues identified throughout the rest of the website have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 2.4.2 Page Titled (Level A)](https://www.w3.org/TR/WCAG21/#page-titled)
* [WCAG 2.1 Success Criterion 4.1.1 Parsing (Level A)](https://www.w3.org/TR/WCAG21/#parsing)

## Test data
Test date: January 2021

**Comment by gmajoulet** [comment]
Closing as this is related to our testing documents and doesn't reflect what happens in production with publisher created content.
